engine/src/main/java/org/terasology/rendering/nui/widgets/UIText.java

```diff
@@ -177,6 +180,15 @@ public UIText(String id) {
         cursorTexture = Assets.getTexture("engine:white").get();
     }
 
+    private String buildPasswordString() {
```

Comment:
I was trying to figure out a better approach than using the StringBuilder but nothing came to my mind. Will now replace with your snippet, thanks.